community/cypher/cypher-compiler-2.3/src/main/java/org/neo4j/cypher/internal/compiler/v2_3/birk/CompiledMathHelper.java

```diff
@@ -51,12 +52,39 @@ public static Object add( Object lhs, Object rhs )
             return String.valueOf( lhs ) + String.valueOf( rhs );
         }
 
+        //List addition
+        if (lhs instanceof List<?> && rhs instanceof List<?>)
+        {
+            List<?> lhsList = (List<?>) lhs;
+            List<?> rhsList = (List<?>) rhs;
+            List<Object> result = new ArrayList<>( lhsList.size() + rhsList.size() );
+            result.addAll( lhsList );
+            result.addAll( rhsList );
+            return result;
+        }
+        else if (lhs instanceof List<?>)
+        {
+            List<?> lhsList = (List<?>) lhs;
+            List<Object> result = new ArrayList<>( lhsList.size() + 1 );
+            result.addAll( lhsList );
+            result.add(rhs);
+            return result;
+        }
+        else if (rhs instanceof List<?>)
+        {
+            List<?> rhsList = (List<?>) rhs;
+            List<Object> result = new ArrayList<>( rhsList.size() + 1 );
+            result.add(lhs);
```

Comment:
Code style for `if` and method invocation should have padding space inside parenthesis (four occurrences above this comment). Not sure about casts; are they with or without padding?